### PR TITLE
Bug fixes ld

### LINKED ISSUE #170: [PE-D] Stray line when all animes are in a certain status
-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: AllardQuek/ped#20

### PR DESCRIPTION
**Pull request by luodan01**
Fix #170, #110 
Add tests for `Stats` class

### CODE REVIEW

File: src/main/java/seedu/anilist/ui/StatsDisplay.java [hunk 1]
```diff
@@ -57,6 +58,28 @@
...
+
+        public IntegerStringConverter() {}
+
+        @Override
+        public String toString(Number object) {
+            String tickLabel = "";
+            //handle error where integer values do not show up due to rounding errors
+            if (Math.ceil(object.doubleValue()) == (Math.round(object.doubleValue() * mult) / mult)) {
+                tickLabel += (int) Math.ceil(object.doubleValue());
+            }
```
Comment (NekrozQliphort) [inline src/main/java/seedu/anilist/ui/StatsDisplay.java:72]: Might I suggest this way instead (define `epsilon` on top):
```suggestion
            if (object.intValue() - epsilon <= object.doubleValue()
                && object.intValue() + epsilon >= object.doubleValue()) {
                tickLabel += object.intValue();
            } else if (object.intValue() - epsilon + 1 <= object.doubleValue()
                && object.intValue() + epsilon + 1 >= object.doubleValue()) {
                tickLabel += (object.intValue() + 1);
            }
```

File: src/main/java/seedu/anilist/ui/StatsDisplay.java [hunk 2]
```diff
@@ -57,6 +58,28 @@
     @FXML
     private StackPane statusbarPlaceholder;
 
+    class IntegerStringConverter extends StringConverter<Number> {
+        private long mult = 100000000;
```
Comment (NekrozQliphort) [inline src/main/java/seedu/anilist/ui/StatsDisplay.java:62]: Maybe use this in conjunction to the method proposed below:
```suggestion
        private final double epsilon = 1E-3;
```

**Review by NekrozQliphort — COMMENTED** [review]
LGTM except maybe a few suggestions

File: src/main/java/seedu/anilist/ui/StatsDisplay.java (hunk 2 above)
Comment (luodan01) [reply src/main/java/seedu/anilist/ui/StatsDisplay.java:62]: thanks I think your method alone is sufficient

**Review by NekrozQliphort — APPROVED** [review]
LGTM!

**Pull request merged**